### PR TITLE
Adding a capture auth idempotent function

### PR DESCRIPTION
**Pull request by kenbolt**
Being able to use idempotency: https://developer.paypal.com/docs/api-basics/#api-idempotency

### CODE REVIEW

File: order.go [hunk 1]
```diff
@@ -95,6 +95,31 @@ func (c *Client) CaptureOrder(orderID string, captureOrderRequest CaptureOrderRe
...
 
+// CaptureOrder with idempotency - https://developer.paypal.com/docs/api/orders/v2/#orders_capture
+// Endpoint: POST /v2/checkout/orders/ID/capture
+// https://developer.paypal.com/docs/api/reference/api-requests/#http-request-headers
+func (c *Client) CaptureIdempotentOrder(
+	orderID string,
+	captureOrderRequest CaptureOrderRequest,
+	requestID string,
+) (*CaptureOrderResponse, error) {
+	capture := &CaptureOrderResponse{}
```
Comment (roopakv) [inline order.go:94]: thank you for the PR. Could you please make the `CaptureOrder` method above use this method? Seems like the only difference is setting the `PayPal-Request-Id` header?

Comment (kenbolt) [reply order.go:94]: sure!

File: order.go [hunk 2]
```diff
@@ -80,6 +80,17 @@ func (c *Client) AuthorizeOrder(orderID string, authorizeOrderRequest AuthorizeO
 // CaptureOrder - https://developer.paypal.com/docs/api/orders/v2/#orders_capture
 // Endpoint: POST /v2/checkout/orders/ID/capture
 func (c *Client) CaptureOrder(orderID string, captureOrderRequest CaptureOrderRequest) (*CaptureOrderResponse, error) {
+	return c.CaptureIdempotentOrder(orderID, captureOrderRequest, "")
+}
+
+// CaptureOrder with idempotency - https://developer.paypal.com/docs/api/orders/v2/#orders_capture
+// Endpoint: POST /v2/checkout/orders/ID/capture
+// https://developer.paypal.com/docs/api/reference/api-requests/#http-request-headers
+func (c *Client) CaptureIdempotentOrder(
```
Comment (roopakv) [inline order.go]:
```suggestion
func (c *Client) CaptureOrderWithPaypalRequestId(
```

File: order.go [hunk 3]
```diff
@@ -80,6 +80,17 @@ func (c *Client) AuthorizeOrder(orderID string, authorizeOrderRequest AuthorizeO
 // CaptureOrder - https://developer.paypal.com/docs/api/orders/v2/#orders_capture
 // Endpoint: POST /v2/checkout/orders/ID/capture
 func (c *Client) CaptureOrder(orderID string, captureOrderRequest CaptureOrderRequest) (*CaptureOrderResponse, error) {
+	return c.CaptureIdempotentOrder(orderID, captureOrderRequest, "")
```
Comment (roopakv) [inline order.go]:
```suggestion
	return c.CaptureOrderWithPaypalRequestId(orderID, captureOrderRequest, "")
```

**Review by roopakv — APPROVED** [review]

**Pull request merged**